File: src/frontend/simulador-investimentos-web/src/app/core/models/cdb-calculation.ts
/** Contrato de entrada da Web API (POST /api/v1/cdb/calculations). */
export interface CalculateCdbRequest {
  /** Valor monetário positivo a ser aplicado. */
  initialAmount: number;
  /** Prazo de resgate em meses, maior que 1. */
  months: number;
}

/** Contrato de saída da Web API. */
export interface CalculateCdbResponse {
  /** Resultado bruto do investimento. */
  grossAmount: number;
  /** Resultado líquido do investimento. */
  netAmount: number;
  /** Alíquota de IR aplicada, em forma decimal (0.225 = 22,5%). */
  incomeTaxRate: number;
  /** Valor do imposto retido. */
  incomeTaxAmount: number;
}

File: src/backend/SimuladorInvestimentos.Domain/Cdb/ValueObjects/CdbCalculation.cs
namespace SimuladorInvestimentos.Domain.Cdb.ValueObjects;

/// <summary>
/// Resultado de um cálculo de CDB: o que o investimento rendeu (bruto), o imposto e o líquido.
/// </summary>
/// <param name="InitialAmount">Valor inicial aplicado.</param>
/// <param name="Months">Prazo em meses usado no cálculo.</param>
/// <param name="GrossAmount">Resultado bruto (VF ao final do último mês).</param>
/// <param name="IncomeTaxRate">Alíquota de IR aplicada, em forma decimal (22,5% => 0.225).</param>
/// <param name="IncomeTaxAmount">Imposto retido sobre o rendimento.</param>
/// <param name="NetAmount">Resultado líquido (bruto menos imposto).</param>
public sealed record CdbCalculation(
    decimal InitialAmount,
    int Months,
    decimal GrossAmount,
    decimal IncomeTaxRate,
    decimal IncomeTaxAmount,
    decimal NetAmount);
